tests/integration/pilot/multiplecontrolplanes/main_test.go

```diff
@@ -0,0 +1,264 @@
+values:
+  global:
+    istioNamespace: %s
+  pilot:
+    env:
+      # CR scoping is guarded by the feature flag
+      ENABLE_ENHANCED_RESOURCE_SCOPING: true`,
+				userGroup1NS.Name(), userGroup1NS.Name())
+		})).
+		Setup(istio.Setup(nil, func(ctx resource.Context, cfg *istio.Config) {
+			s := ctx.Settings()
+			// TODO test framework has to be enhanced to use istioNamespace in istioctl commands used for VM config
+			s.SkipWorkloadClasses = append(s.SkipWorkloadClasses, echo.VM)
+
+			cfg.Values["global.istioNamespace"] = userGroup2NS.Name()
+			cfg.SystemNamespace = userGroup2NS.Name()
+			cfg.ControlPlaneValues = fmt.Sprintf(`
+namespace: %s
+revision: usergroup-2
+meshConfig:
+  # REGISTRY_ONLY is used to verify custom resources scoping
+  outboundTrafficPolicy:
+    mode: REGISTRY_ONLY
+  # CR scoping requires discoverySelectors to be configured
+  discoverySelectors:
+    - matchLabels:
+        usergroup: usergroup-2
+values:
+  global:
+    istioNamespace: %s
+  pilot:
+    env:
+      # CR scoping is guarded by the feature flag
+      ENABLE_ENHANCED_RESOURCE_SCOPING: true`,
+				userGroup2NS.Name(), userGroup2NS.Name())
+		})).
+		SetupParallel(
+			// application namespaces are labeled according to the required control plane ownership.
+			namespace.Setup(&echo1NS, namespace.Config{Prefix: "echo1", Inject: true, Revision: "usergroup-1", Labels: map[string]string{"usergroup": "usergroup-1"}}),
+			namespace.Setup(&echo2NS, namespace.Config{Prefix: "echo2", Inject: true, Revision: "usergroup-2", Labels: map[string]string{"usergroup": "usergroup-2"}}),
+			namespace.Setup(&echo3NS, namespace.Config{Prefix: "echo3", Inject: true, Revision: "usergroup-2", Labels: map[string]string{"usergroup": "usergroup-2"}})).
+		SetupParallel(
+			deployment.Setup(&apps, deployment.Config{
+				Namespaces: []namespace.Getter{
+					namespace.Future(&echo1NS),
+					namespace.Future(&echo2NS),
+					namespace.Future(&echo3NS),
+				},
+			})).
+		Run()
+}
+
+// TestMultiControlPlane sets up two distinct istio control planes and verify if resources and traffic are properly isolated
+func TestMultiControlPlane(t *testing.T) {
+	framework.NewTest(t).
+		Features("installation.multiplecontrolplanes").
+		Run(func(t framework.TestContext) {
+			// configure peerauthentication per system namespace
+			restrictUserGroups(t)
+
+			testCases := []struct {
+				name       string
+				statusCode int
+				from       echo.Instances
+				to         echo.Instances
+			}{
+				{
+					name:       "workloads within same usergroup can communicate, same namespace",
+					statusCode: http.StatusOK,
+					from:       apps.NS[0].A,
+					to:         apps.NS[0].B,
+				},
+				{
+					name:       "workloads within same usergroup can communicate, different namespaces",
+					statusCode: http.StatusOK,
+					from:       apps.NS[1].A,
+					to:         apps.NS[2].B,
+				},
+				{
+					name:       "workloads within different usergroups cannot communicate",
```

Comment:
nit: "cannot communicate" is a bit misleading; this is actually the client blocking itself. We may want to have a test that asserts the server denies the client as well, if it manages to make a request though

Reply:
i added another test for case where default passthrough is present, and the server rejects the request

Reply:
Basically, removed REGISTRY_ONLY setting from one of the control planes, so that the traffic can be tested with passthrough cluster